scripts/initialize.sh

```diff
@@ -72,9 +76,46 @@ while getopts "n:o:p:d:b:a:s:k:h" opt; do
 done
 shift $(expr $OPTIND - 1) # remove options from positional parameters
 
+CONFIG_DIR=${config_dir:-$CONFIG_DIR_DEFAULT}
+
+# read from the config file if a stake_owner is not provided as parameter
 if [ -z "$stake_owner" ]; then
-   echo 'Stake owner address must be provided. See --help'
-   exit 1
+   printf "\nReading stake owner address from the config file..."
+
+   # read from the config file when the stake owner is not provided
+   config_files=($CONFIG_DIR/*.toml)
+   config_files_count=${#config_files[@]}
+   while :; do
+      printf "\nSelect client config file: \n"
+      i=1
+      for o in "${config_files[@]}"; do
+         echo "$i) ${o##*/}"
+         let i++
+      done
+
+      read reply
+      if [ "$reply" -ge 1 ] && [ "$reply" -le $config_files_count ]; then
+         CONFIG_FILE_PATH=${config_files["$reply" - 1]}
+         break
+      else
+         printf "\nInvalid choice. Please choose an existing option number.\n"
+      fi
+   done
+
+   sed -n -e '/^keyFile -/p' $CONFIG_FILE_PATH
+
+   key_file_str=$(grep "^keyFile" $CONFIG_FILE_PATH)
```

Comment:
Could we also add some prints to help debugging in case of problems?

```sh
   key_file_path=$(awk '/keyFile ?=(.*)$/{print $3}' $CONFIG_FILE_PATH | xargs)

   printf "Reading an address from the key file: ${key_file_path}\n"

   stake_owner=$(jq -jr .address $key_file_path)

   printf "Discovered the address: ${stake_owner}\n"
```

Reply:
Sure thing [ac7cf84](https://github.com/keep-network/keep-core/pull/3163/commits/ac7cf8493cb9164edaf194a3421b5f922845c111)